File: GavApp/Sources/GavApp/Domain/Message/Models/Message.swift
import Foundation

public struct Message: Identifiable, Codable {
    public let id: UUID
    public let chatId: UUID
    public let senderId: UUID
    public let text: String? // может быть nil, если attachment
    public let replyToId: UUID?
    public let createdAt: Date
    public var editedAt: Date?

    public var attachments: [Attachment] = []
    public var reactions: [Reaction] = []
}

File: backend/messenger/transport/http/dto/message.go
package dto

import (
	"github.com/google/uuid"
	"time"
)

type MessageResponseDTO struct {
	ID       uuid.UUID `json:"id"`
	ChatID   uuid.UUID `json:"chat_id"`
	SenderID uuid.UUID `json:"sender_id"`

	Text      *string    `json:"text,omitempty"`
	ReplyToID *uuid.UUID `json:"reply_to_id,omitempty"`

	CreatedAt time.Time  `json:"created_at"`
	EditedAt  *time.Time `json:"edited_at,omitempty"`
}
